project-base/app/tests/FrontendApiBundle/Functional/Blog/Category/BlogCategoryTest.php

```diff
@@ -225,6 +226,51 @@ public function testGetBlogCategoryReturnsErrorWithWrongSlug(): void
         $this->assertSame($expectedErrorMessage, $errors[0]['message']);
     }
 
+    /**
+     * @param string $slug
+     * @param string|null $expectedImage
+     */
+    #[DataProvider('getBlogCategoryImageDataProvider')]
+    public function testGetBlogCategoryImage(string $slug, ?string $expectedImage): void
+    {
+        $query = '
+            query {
+                blogCategory(urlSlug: "' . $slug . '") {
+                    mainImage {
+                        url
+                    }
+                }
+            }
+        ';
+
+        $response = $this->getResponseContentForQuery($query);
+        $data = $this->getResponseDataForGraphQlType($response, 'blogCategory');
+
+        $this->assertArrayHasKey('mainImage', $data);
+
+        if ($expectedImage === null) {
+            $this->assertNull($data['mainImage']);
+        } else {
+            $this->assertStringEndsWith($expectedImage, $data['mainImage']['url']);
+        }
+    }
+
+    /**
+     * @return iterable
+     */
+    public static function getBlogCategoryImageDataProvider(): iterable
+    {
+        yield [
+            'slug' => 'first-subsection-en',
```

Comment:
maybe we can use the persistent reference names and then fetch the category by uuid rather than slug in the test? I am afraid this way the test fails when the first domain locale is changed on a project :frowning_face:

Reply:
good thinking :) thanks, changed